### PR TITLE
Revise the Snapcraft packaging

### PR DESCRIPTION
**Pull request by olof-nord**
* Add link to go and nodejs snapcraft plugin documentation
* Remove unused build environment variables
* Remove no longer needed build dependencies
* Switch to [npm ci](https://docs.npmjs.com/cli/ci) instead of npm install
* Do not use global npm installs without version, but use the package.json file to have all dependencies versioned

The changed npm dependency management ensures that each build will use exactly the same package versions.
With `npm ci`, the package-lock.json file is used, which specifies exactly which package versions to use (instead of the approximate version, which is what the package.json file does)
The global installs are also removed, as these dependencies are already present in package.json. Installing without specifying the version means each build could potentially get a different version.

Future:
Once/if https://github.com/nanu-c/axolotl/pull/195 is merged, remove snap-specific backend code in favour of environment variables and parameters, and also add documentation regarding `snap` and `snapcraft` usage to docs/INSTALL.md.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -1,67 +1,60 @@
 name: axolotl
-summary: An signal client.
+summary: A signal client.
 description: |
   Axolotl is a crossplattform signal client
```
Comment (Flaburgan) [inline snap/snapcraft.yaml:4]: If you're fixing typo, let's do it there too ;)

Comment (olof-nord) [reply snap/snapcraft.yaml:4]: You mean with the description?
I dont see an issue there, what is your proposal?

Comment (Flaburgan) [reply snap/snapcraft.yaml:4]: there should be only one t to platform, shouldn't it?

Comment (olof-nord) [reply snap/snapcraft.yaml]: you got it. Didnt notice before

**Comment by Flaburgan** [comment]
You need to rebase this branch now that 0.8.8 is merged.

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -1,67 +1,60 @@
...
       - libgconf-2-4
       - libnss3
       - libx11-xcb1
       - libxss1
       - libxtst6
-    override-build: |
-      echo "This runs before build!"
-      snapcraftctl build
+
   axolotl-web:
```
[the remark this answers is not in the document]
Comment (olof-nord) [reply snap/snapcraft.yaml:29]: I didnt come across this, but I noticed that snap caches as much as possible during build.
Do you get the same error if you go `sudo snapcraft clean` before `sudo snapcraft`?

[the remark this answers is not in the document]
Comment (olof-nord) [reply snap/snapcraft.yaml:29]: Either way, lets add `git` as a build dependeny

[the remark this answers is not in the document]
Comment (nanu-c) [reply snap/snapcraft.yaml:29]: Yeah lets add git

**Comment by Flaburgan** [comment]
@olof-nord can you please rebase that branch on `main` so we can merge it?

File: snap/snapcraft.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (olof-nord) [reply snap/snapcraft.yaml:29]: Added

**Pull request merged**